pkg/kubectl/cmd/rollout/rollout_restart.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rollout
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/printers"
+	"k8s.io/cli-runtime/pkg/resource"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/scheme"
+	"k8s.io/kubernetes/pkg/kubectl/util/i18n"
+	"k8s.io/kubernetes/pkg/kubectl/util/templates"
+)
+
+// RestartOptions is the start of the data required to perform the operation.  As new fields are added, add them here instead of
+// referencing the cmd.Flags()
+type RestartOptions struct {
+	PrintFlags *genericclioptions.PrintFlags
+	ToPrinter  func(string) (printers.ResourcePrinter, error)
+
+	Resources []string
+
+	Builder          func() *resource.Builder
+	Namespace        string
+	EnforceNamespace bool
+
+	resource.FilenameOptions
+	genericclioptions.IOStreams
+}
+
+var (
+	restartLong = templates.LongDesc(`
+		Restart a resource.
+
+	        A deployment with the "RolloutStrategy" will trigger a rolling restart.`)
+
+	restartExample = templates.Examples(`
+		# Restart a deployment
+		kubectl rollout restart deployment/nginx`)
+)
+
+// NewRolloutRestartOptions returns an initialized RestartOptions instance
+func NewRolloutRestartOptions(streams genericclioptions.IOStreams) *RestartOptions {
+	return &RestartOptions{
+		PrintFlags: genericclioptions.NewPrintFlags("restarted").WithTypeSetter(scheme.Scheme),
+		IOStreams:  streams,
+	}
+}
+
+// NewCmdRolloutRestart returns a Command instance for 'rollout restart' sub command
+func NewCmdRolloutRestart(f cmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewRolloutRestartOptions(streams)
+
+	validArgs := []string{"deployment"}
+
+	cmd := &cobra.Command{
+		Use:                   "restart RESOURCE",
+		DisableFlagsInUseLine: true,
+		Short:                 i18n.T("Restart a paused resource"),
```

Comment:
Remove "pause" otherwise this will be confused with `kubectl rollout resume`

Reply:
bad copy paste ;-)